CONTRIBUTING.md

```diff
@@ -1,49 +1,122 @@
 # OSML Contributing Guidelines
 
 Thank you for your interest in contributing to our project! This document will guide you through the process of
-contributing to our repository using a release-based GitFlow workflow. Please read the guidelines carefully in order to
+contributing to our repository using a release-based GitFlow workflow. Please read the guidelines carefully to
 submit your pull requests effectively.
 
 ## Table of Contents
 
-- [Release-based GitFlow](#release-based-gitflow)
-- [Branches](#branches)
+- [Trunk-Based GitFlow](#trunk-based-gitflow)
 - [Linting](#linting)
-- [Use Cases](#use-cases)
-   - [Develop a new feature](#develop-a-new-feature)
-   - [Develop multiple features in parallel](#develop-multiple-features-in-parallel)
-   - [Create and deploy a release](#create-and-deploy-a-release)
-   - [Production hot fix](#production-hot-fix)
 - [Code Style](#code-style)
 - [Commit Messages](#commit-messages)
 - [Issue Tracking](#issue-tracking)
 
-## Release-based GitFlow
+## Trunk-Based GitFlow
 
-We follow a release-based GitFlow branching model to manage our codebase. It involves different types of branches to
-ensure proper version control and collaboration. The primary branches in this workflow are:
+![Trunk-Based GitFlow](./images/tbd.png)
 
-note- `main`: This branch represents the latest production-ready state of the project code.
-- `dev`: This branch is used as the integration branch, combining all feature branches for a specific release.
-- `release/vX.Y.Z`: Branches in the format of `release/vX.Y.Z` are used to prepare a specific release.
-- `feature/*`: Feature branches are created for the development of new features or significant enhancements.
-- `hotfix/*`: Hotfix branches are created for bug fixes against production code.
+We follow a trunk-based GitFlow branching model to manage our codebase.
+It involves different types of branches to ensure proper version control and collaboration.
+This guide provides a step-by-step example of implementing a Trunk-Based Development (TBD) workflow,
+covering feature development, rebasing, and the hotfix process.
+The primary branches developers create in this workflow are:
```

Comment:
You're confusing conventional commit types with branch types. They're not the same thing. The only thing that changes with a scaled trunk based development is we're removing the `dev` branch. Feature branches and release branches are still a thing. We can likely do without the `hotfix` branch. Ideally the feature branches are short-lived and deleted after each PR so the name isn't critical but keeping the feature/ convention isn't bad.